server/src/infra/repositories/asset.repository.ts

```diff
@@ -761,7 +761,7 @@ export class AssetRepository implements IAssetRepository {
     let builder = this.repository
       .createQueryBuilder('asset')
       .where('asset.isVisible = true')
-      .andWhere('asset.fileCreatedAt < NOW()');
+      .andWhere("asset.fileCreatedAt BETWEEN '4713-01-01 BC' AND '294276-12-31 AD'");
```

Comment:
Hmm, this seems a little insane lol. What if we just did less than 9999?

Reply:
yeah it is a bit crazy haha.

We could just do less than 9999, but we would need a lower constraint on it too since it could also be out of range in that direction.

**Counter argument though:**
Allowing such a wide range would allow users to view any assets with weird dates and be able to change those dates through the UI manually. It would make troubleshooting easier and potentially prevent future related issues.

Reply:
> it could also be out of range

Maybe, but it hasn't happened yet at least. If we are going to change this query to using `BETWEEN` can you show the results of running an `EXPLAIN ANALYZE` on the resulting query? This is a pretty critical query and we should check that it doesn't introduce any performance impact for instances that have lots of assets. Alternatively, we already know that less than DATETIME already works fine.

Reply:
Let's rebase this after #6156 is merged so we can get a nice delta for the generated SQL.

Reply:
> What if we just did less than 9999

I think we've seen a few broken assets with 5-digit years in the past.